doc/developer-guide/plugins/remap-plugins.en.rst

```diff
@@ -91,18 +92,30 @@ If there is an error then the callback should return :macro:`TS_ERROR` and fill
 Configuration reload notifications
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+Most of the plugins is assumed to use per-plugin-instance data-structures when reloading their
+configs and only a few of them that wish to optimize performance or deal with the complexities
+of using a per-plugin DSO "global" data-structures would use pluging configuration reload
+notifications like :func:`TSRemapPreConfigReload` and :func:`TSRemapPostConfigReload`.
+
+Instead of trying to foresee the needs or the expectations of each use-case, a more "open-ended"
+and straight-forward design was chosen for the configuration reload notifications.
+The notifications are broadcasted to all loaded plugins at the moments before and after 
+the reload attempt, regardless of whether a plugin is part of the new configuration or not.
+
 :func:`TSRemapPreConfigReload` is called *before* the parsing of a new remap configuration starts
 to notify plugins of the coming configuration reload. It is called on all already loaded plugins,
 invoked by current and all previous still used configurations. This is an optional entry point.
 
 :func:`TSRemapPostConfigReload` is called to indicate the end of the the new remap configuration
 load. It is called on the newly and previously loaded plugins, invoked by the new, current and
-previous still used configurations. It also indicates if the configuration reload was successful
-by passing :macro:`TS_SUCCESS` or :macro:`TS_ERROR`. This is an optional entry point.
+previous still used configurations. It also indicates wheather the configuration reload was successful
+by passing :macro:`TSREMAP_CONFIG_RELOAD_FAILURE` in case of failure and to notify the plugins if they
+are going to be part of the new configuration by returning :macro:`TSREMAP_CONFIG_RELOAD_SUCCESS_PLUGIN_USED`
```

Comment:
"returning"? See previous comments on similar text earlier.

Reply:
right, passed to the plugin